### PR TITLE
[BE] Use `C10_CLANG_DIAGNOSTIC` macros

### PR DESCRIPTION
**Pull request by malfet**
Fixes #ISSUE_NUMBER

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

File: c10/util/Half.h [hunk 1]
```diff
@@ -428,10 +428,9 @@ struct alignas(4) complex<Half> {
 // The overflow checks may involve float to int conversion which may
 // trigger precision loss warning. Re-enable the warning once the code
 // is fixed. See T58053069.
-#ifdef __clang__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunknown-warning-option"
-#pragma GCC diagnostic ignored "-Wimplicit-int-float-conversion"
+C10_CLANG_DIAGNOSTIC_PUSH()
```
Comment (kit1980) [inline c10/util/Half.h:431]: It was "GCC diagnostic push", but now "clang diagnostic push", is it OK?

Comment (malfet) [reply c10/util/Half.h:431]: Yes, for `clang` those are aliases, and it was guarded under `#ifdef __clang__`

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**